File: frontend/src/lib/tenant.ts
# frontend/src/lib/tenant.ts
/**
 * Tenant context and helpers
 */

export interface Tenant {
  id: string;
  name: string;
  industry: "immigration" | "healthcare";
  subdomain: string;
  plan: "starter" | "professional" | "enterprise";
  is_active: boolean;
  brand_name?: string;
  brand_logo_url?: string;
  brand_primary_color?: string;
  settings: Record<string, any>;
}

export function getIndustryLabel(industry: string): string {
  return industry === "immigration" ? "Immigration & Recruitment" : "Healthcare";
}

export function getIndustryIcon(industry: string): string {
  return industry === "immigration" ? "✈️" : "🏥";
}

export function getPlanLabel(plan: string): string {
  const labels: Record<string, string> = {
    starter: "Starter",
    professional: "Professional",
    enterprise: "Enterprise",
  };
  return labels[plan] || plan;
}

export function getPlanColor(plan: string): string {
  const colors: Record<string, string> = {
    starter: "bg-gray-100 text-gray-800",
    professional: "bg-blue-100 text-blue-800",
    enterprise: "bg-purple-100 text-purple-800",
  };
  return colors[plan] || "bg-gray-100 text-gray-800";
}

File: backend/app/models/tenant.py
# backend/app/models/tenant.py
"""
Tenant Model - Multi-tenant organization entity
Each tenant represents a firm/clinic using EXL-CRM
"""

from beanie import Document
from pydantic import Field
from datetime import datetime
from typing import Optional, List
from enum import Enum


class IndustryType(str, Enum):
    IMMIGRATION = "immigration"
    HEALTHCARE = "healthcare"


class SubscriptionPlan(str, Enum):
    STARTER = "starter"
    PROFESSIONAL = "professional"
    ENTERPRISE = "enterprise"


class Tenant(Document):
    """Tenant organization entity."""

    name: str = Field(..., min_length=1, max_length=200)
    industry: IndustryType
    subdomain: str = Field(..., min_length=3, max_length=63, pattern=r"^[a-z0-9-]+$")
    plan: SubscriptionPlan = SubscriptionPlan.STARTER
    is_active: bool = True
    custom_domain: Optional[str] = None

    # Settings stored as flexible dict
    settings: dict = Field(default_factory=dict)

    # White-label branding
    brand_name: Optional[str] = None
    brand_logo_url: Optional[str] = None
    brand_primary_color: Optional[str] = "#2563eb"

    created_at: datetime = Field(default_factory=datetime.utcnow)
    updated_at: datetime = Field(default_factory=datetime.utcnow)

    class Settings:
        name = "tenants"
        indexes = [
            "subdomain",
            "industry",
            "plan",
        ]
